Userland/Applications/SystemMonitor/main.cpp

```diff
@@ -354,6 +354,8 @@ int main(int argc, char** argv)
     };
 
     process_table_view.on_selection_change = [&] {
+        if (!process_table_view.is_focused())
+            return;
```

Comment:
Hmm, what makes sure that we update the actions once we *do* get focus?

Reply:
Since we need to select a row to interact with the table view it will always have been in focus when something is selected, but there could be some cases where it doesn't work as intended. And I have to agree it's a bit blunt to use is_focused() for something like this.

I'll change it to comparing with a static variable so we never check permissions on the same pid twice in a row. I'll also add a check so that we don't send to pid -1 or 0.